Sólo en mv-patched/: mv.bak
diff -u mv/mv.c mv-patched/mv.c
--- mv/mv.c	2026-02-16 10:32:03.709741356 -0500
+++ mv-patched/mv.c	2026-02-16 10:39:22.267183699 -0500
@@ -355,9 +355,11 @@
 	}
 	if (fchmod(to_fd, sbp->st_mode))
 		warn("%s: set mode", to);
+	#ifdef __linux__
+	#else
 	if (fchflags(to_fd, sbp->st_flags) && (errno != EOPNOTSUPP))
 		warn("%s: set flags (was: 0%07o)", to, sbp->st_flags);
-
+	#endif
 	if (close(to_fd)) {
 		warn("%s", to);
 		return (1);
