diff --color -u bin/ksh/trap.c ksh/trap.c
--- trap.c	2026-02-18 11:17:13.898798551 -0500
+++ trap.c	2026-02-18 11:22:51.511295406 -0500
@@ -32,11 +32,11 @@
 	extern char	*sys_siglist[];
 # endif
 	int	i;
-
+	const char *s = strsignal(i);
 	/* Use system description, if available, for unknown signals... */
 	for (i = 0; i < NSIG; i++)
-		if (!sigtraps[i].name && sys_siglist[i] && sys_siglist[i][0])
-			sigtraps[i].mess = sys_siglist[i];
+		if (!sigtraps[i].name && s && s[0])
+			sigtraps[i].mess = s;
 #endif	/* HAVE_SYS_SIGLIST */
 
 	sigemptyset(&Sigact_ign.sa_mask);
